### ISSUE TITLE
cpuset config: cpuset config is wrong on i3.2xlarge

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#3911]
*Installation details*
Scylla version (or git commit hash): 666.development-0.20181111.1c5934c93
Cluster size: 1
OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-0eab544d3f4de5cdd
Instance type: i3.2xlarge

```
# cat /etc/scylla.d/cpuset.conf |grep -v \#
CPUSET="--cpuset 1-7 "
```

Expected: 
```
CPUSET="--cpuset 0-7 "
```

**Comment by glommer** [comment]
There is nothing wrong with `0-7` if perftune.yaml is in `mq` mode.
i3.2xlarge has 8 Queues, so we should be able to do mq just fine.

**Comment by amoskong** [comment]
so 0-7  or  1-7 are all correct?

**Comment by glommer** [comment]
None of them, by themselves, are bad.

If we use 0-7 but perftune.yaml says mode `sq` or `sq_split`, that is bad (if it says `mq` it is fine).
If we use 1-7 but perftune says mode `sq_split` or `mq`, that is bad (if it says `sq` it is fine)

**Comment by tzach** [comment]
> If we use 0-7 but perftune.yaml says mode sq or sq_split, that is bad (if it says mq it is fine).
> If we use 1-7 but perftune says mode sq_split or mq, that is bad (if it says sq it is fine)

@amoskong which is it?
Is this a change from 2.3?

**Comment by amoskong** [comment]
@tzach Yes, it's always 0-7 in 2.3

**Comment by slivne** [comment]
@glommer  / @vladzcloudius 

so 2.3 has 0-7 
master/3.0 have 1-7 

was this change intentional ... if so we can close the issue - if not lets fix it.

**Comment by glommer** [comment]
I don't know whether it was intentional or not - @vladzcloudius ?
However, nobody has answered me yet if we have `sq` in perftune.yaml.

If we do, there is nothing to fix, really. 0-7+mq is not "more right" than 1-7+sq

**Comment by vladzcloudius** [comment]
This result may happen if we run the sciptology that generates `cpuset.conf` before the `nvme`-based mountpoint is mounted. In this case script would (correctly) conclude that `/var/lib/scylla` is mounted on the EBS drive and this device has a single IRQ which yields the `sq` mode on the CPU configuration like on `i3.2xlarge`: 4 or less physical cores.

I'll check our scriptology.

**Comment by vladzcloudius** [comment]
@glommer Of course we have `sq` in perftune.py. We support all 3 modes. Except for HW queues equality to thr number of CPU threads case the logic is:
* Choose mq if number of HT less or equal to 4.
* Else choose sq if number of cores less or equal to 4.
* Else choose sq_split.

**Comment by vladzcloudius** [comment]
Patch with the fix is on the list

**Comment by slivne** [comment]
@vladzcloudius can you please reference the patch 

was the patch committed if not please ping

**Comment by vladzcloudius** [comment]
@slivne "[PATCH scylla-ami v1] ami: scylla_ami_setup: call sysconfig_setup after create_devices"

**Comment by vladzcloudius** [comment]
@slivne pinged

**Comment by slivne** [comment]
was backported to 3.0 as well @amoskong I'll need you to verify its fixed.

**Comment by amoskong** [comment]
@slivne 

The problem had been fixed on master & 3.0

All subtests is PASS in latest scylla-ami-test (both 3.0 & master).
- http://jenkins.cloudius-systems.com:8080/view/scylla-3.0/job/scylla-3.0/job/3.0-test/job/scylla-ami-test/label=muninn/23/  (job failed for jenkins issue)
- http://jenkins.cloudius-systems.com:8080/job/scylla-master/job/ami-tests/label=muninn/14/ (PASS)
